File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class DashboardStats(BaseModel):
    total_jobs: int = 0
    avg_score: float | None = None
    strong_apply_count: int = 0
    apply_count: int = 0
    maybe_count: int = 0
    skip_count: int = 0
    applied_count: int = 0
    interviewing_count: int = 0
    offer_count: int = 0
    response_rate: float = 0.0


class ChartDataPoint(BaseModel):
    label: str
    value: float | int
    color: str | None = None

File: frontend/src/types/analytics.ts
export interface DashboardStats {
  total_jobs: number;
  avg_score: number | null;
  strong_apply_count: number;
  apply_count: number;
  maybe_count: number;
  skip_count: number;
  applied_count: number;
  interviewing_count: number;
  offer_count: number;
  response_rate: number;
}

export interface ChartDataPoint {
  label: string;
  value: number;
  color?: string;
}
